packages/cache/src/internal/tar.ts

```diff
@@ -7,21 +7,18 @@ import {CompressionMethod} from './constants'
 
 const IS_WINDOWS = process.platform === 'win32'
 
-async function getTarPath(
-  args: string[],
-  compressionMethod: CompressionMethod
-): Promise<string> {
+async function getTarPath(args: string[]): Promise<string> {
+  let tarPath = await io.which('tar', true)
   switch (process.platform) {
     case 'win32': {
+      const gnuTar = await utils.getGnuTarPathOnWindows()
       const systemTar = `${process.env['windir']}\\System32\\tar.exe`
-      if (compressionMethod !== CompressionMethod.Gzip) {
-        // We only use zstandard compression on windows when gnu tar is installed due to
-        // a bug with compressing large files with bsdtar + zstd
+      if (gnuTar) {
+        // Use GNUtar as default on windows
         args.push('--force-local')
+        tarPath = gnuTar
```

Comment:
Should we log debug message that tar is being overridden

Reply:
Changed it back to previous way

Reply:
@Phantsure it still makes sense to add info log on which tar is being used.